### PR TITLE
CRAFTERCMS-1797

### PR DESCRIPTION
**Pull request by dejan-brkic**
Implemented new Dependency Service

### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/1797

### CODE REVIEW

File: src/main/java/org/craftercms/studio/impl/v1/content/pipeline/ExtractAssetDependencyProcessor.java [hunk 1]
```diff
@@ -75,25 +75,19 @@ public void process(PipelineContent content, ResultTO result) throws ContentProc
         }
         try {
             if (isCss || isJs || isTemplate) {
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/content/pipeline/ExtractAssetDependencyProcessor.java]: Would it make sense to just say: if the asset is textual, then try to extract dependencies?

File: src/main/java/org/craftercms/studio/impl/v1/service/content/ContentServiceImpl.java [hunk 2]
```diff
@@ -816,13 +855,7 @@ protected void updateDependenciesOnMove(String site, String fromPath, String mov
             String content = getContentAsString(site, movePath);
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/content/ContentServiceImpl.java]: This is a problem. We're loading a piece of content of unknown size to RAM. Let's discuss.

We should also find other places that we load content into RAM and discuss.

File: src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+            }
+        }
+        return dependencyEntities;
+    }
+
+    private String getCleanPath(String path) {
+        return path.replaceAll("//", "/");
+    }
+
+    private void insertDependencies(List<DependencyEntity> dependencyEntities) {
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/service/dependency/DependencyServiceImpl.java]: Rename to insertDependenciesIntoDatabase? (in our continued mission to make clearer method names...)

**Review by sumerjabri — CHANGES_REQUESTED** [review]
Ping me to discuss as needed.

**Comment by sumerjabri** [comment]
@dejan-brkic also, please fix the Codacy issues you can fix.

**Comment by dejan-brkic** [comment]
Fixed code review requests. Fixed codacy issues

**Comment by sumerjabri** [comment]
Please these Codacy issues (click on show more in Codacy to see these):
```
src/main/java/org/craftercms/studio/impl/v1/content/pipeline/ExtractAssetDependencyProcessor.java
Avoid unused imports such as 'java.util.List'
20
import java.util.List;
Avoid unused imports such as 'java.util.regex.Matcher'
21
import java.util.regex.Matcher;
Avoid unused imports such as 'java.util.regex.Pattern'
22
import java.util.regex.Pattern;
src/main/java/org/craftercms/studio/impl/v1/content/pipeline/ExtractDependencyProcessor.java
Avoid unused imports such as 'org.dom4j.Document'
28
import org.dom4j.Document;
Avoid unused imports such as 'java.util.HashMap'
30
import java.util.HashMap;
Avoid unused imports such as 'java.util.Map'
31
import java.util.Map;
Avoid unused imports such as 'java.util.Set'
32
import java.util.Set;
src/main/java/org/craftercms/studio/impl/v1/service/site/SiteServiceImpl.java
Avoid unused local variables such as 'content'.
1430
InputStream content = contentRepository.getContent(site, path);
src/main/java/org/craftercms/studio/impl/v1/service/workflow/WorkflowServiceImpl.java
Avoid unused local variables such as 'id'.
655
String id = site + ":" + user + ":" + request;
```

**Comment by dejan-brkic** [comment]
Codacy fixes

**Pull request merged**